designs/manual-node-provisioning.md

```diff
@@ -0,0 +1,111 @@
+- **state: Stopped:** This is a "cold pool" option for cost savings. The controller provisions a node, allows it to initialize, and then issues a StopInstances API call to the cloud provider. The node does not join the Kubernetes cluster. When capacity is needed, the controller issues a StartInstances call. This is slower than Running but significantly cheaper.
+- **state: Hibernated:** This is the "warm pool" option. The process is similar to Stopped, but the controller issues a HibernateInstances call. This saves the instance's RAM to the root EBS volume, allowing for much faster resumes.
+
+This offers a balance between the speed of Running and the cost of Stopped, though it has more cloud-provider-specific prerequisites. When a pod needs capacity from a Stopped or Hibernated pool, the overprovisioner controller will start the instance, wait for it to join the cluster, and then make it available for scheduling.
+
+
+### Strategies Explained
+The strategy field provides high-level control over the behavior of the overprovisioning logic.
+
+#### KeepMinimum Strategy: 
+This strategy ensures that a minimum number of spare nodes are always available, regardless of the cluster's overall utilization. It is ideal for:  
+    - **Predictable Bursts:** Pre-scaling for known events where immediate capacity is non-negotiable.
+
+When using KeepMinimum, the sizing.count field is most appropriate, providing a deterministic floor of spare capacity.
+
+
+#### KeepEmpty Strategy (Proportional Overprovisioning): 
+This strategy is designed to maintain a buffer of spare capacity that is proportional to the current size of the cluster, but it allows this buffer to scale down to zero if the cluster becomes completely empty. It is ideal for:  
+    - **Handling Bursts:** Providing headroom for stateless applications to scale horizontally without waiting for new nodes.  
+    - **Cost Efficiency:** Avoiding the cost of maintaining a buffer during periods of no activity. The buffer grows and shrinks with the cluster's usage.  
+
+When using KeepEmpty, the sizing.percentage field is most appropriate, as it creates a dynamic buffer that adapts to the cluster's scale.  
+The controller will use `ceil()` to round up, ensuring at least one spare node is created for any non-zero percentage in a running cluster.
+
+## Alternatives Considered
+
+One of the alternatives suggested was to extend the NodePool API directly[1]. This was rejected because it conflates two distinct responsibilities: defining the properties of a pool of capacity (NodePool) and defining the proactive management strategy for that capacity (Overprovisioner). A separate CRD provides a cleaner separation of concerns, a more intuitive user experience, and avoids creating an overly complex, monolithic API object. The common "pause pod" workaround was also considered and rejected as it is an operationally burdensome hack and not cost effective solve for larger spike.
+
```

Comment:
Was the nodeclaim CRD considered? You can provision nodes by just applying nodeclaim CRs to the cluster. Can you write up an option about that and making the nodeclaim crd manual provisioning a first class concept?

Reply:
@Bryce-Soghigian added info on the nodeclaim crd now. https://github.com/kubernetes-sigs/karpenter/pull/2397/files#diff-69f6e618d321eef3aa550b3e1c9f6547345c899c539c3e45c7f8421758cc9045R91 

it would have a label by which we can identify the nodes to be consolidate or to use for next scaleup.